lib/hls/hls_parser.js

```diff
@@ -2578,7 +2775,11 @@ shaka.hls.HlsParser = class {
       await this.update();
 
       // This may have converted to VOD, in which case we stop updating.
-      if (this.isLive_()) {
+      // Note that all manifests start out listed as VOD, since we won't know if
+      // a manifest is live or not until we parse at least one media playlist.
+      // So we should also keep the update timer going until we at least know
+      // if the manifest is actually VOD or not.
+      if (this.isLive_() || !this.hasEnoughInfoToFinalizeStreams_()) {
```

Comment:
I'm having a hard time tracing this to understand if it can work.

The first playlist update is kicked off by start(), and only if this.updatePlaylistDelay_ is non-zero. 
 After that, onUpdate_() will trigger subsequent updates.

this.updatePlaylistDelay_ is only set in your change inside of determineDuration_(), which is only called from inside createSegmentIndex() after finalizeStreams_().

So it looks as if the initial update() would never happen, and this would never be executed.

Is that accurate?  What did I miss?

Reply:
I think I was intending on making the parser schedule manifest updates for VOD, and then cancel them if the manifest turned out to actually be VOD. But I forgot the first step.
It escaped my notice during testing, I think, because our history of Shaka Player HLS livestream is broken in nightly too, and I assumed I was just seeing that issue. I didn't even notice the manifest wasn't updating, in addition to that!

Anyway, in retrospect it would make a lot more sense to just wait until the finalization step to start scheduling updates. I'll do that instead.